x-pack/elastic-agent/pkg/agent/application/pipeline/emitter/emitter.go

```diff
@@ -25,12 +27,20 @@ func New(ctx context.Context, log *logger.Logger, agentInfo *info.AgentInfo, con
 
 	ctrl := NewController(log, agentInfo, controller, router, modifiers, caps, reloadables...)
 	err := controller.Run(ctx, func(vars []*transpiler.Vars) {
-		ctrl.Set(vars)
+		ctrl.Set(ctx, vars)
 	})
 	if err != nil {
 		return nil, errors.New(err, "failed to start composable controller")
 	}
-	return func(c *config.Config) error {
-		return ctrl.Update(c)
+	return func(ctx context.Context, c *config.Config) (err error) {
+		span, ctx := apm.StartSpan(ctx, "update", "app.internal")
+		defer func() {
+			if err != nil {
+				apm.CaptureError(ctx, err).Send()
+			}
```

Comment:
[Suggestion]
It's safe to call `apm.CaptureError` with a nil error and then call `Send`. Therefore it can be simplified.
```suggestion
		apm.CaptureError(ctx, err).Send()
```